src/run/allocator.rs

```diff
@@ -14,22 +14,39 @@ pub(super) struct Node(pub AtomicU64, pub AtomicU64);
 pub struct Heap(pub(super) [Node]);
 
 impl Heap {
+  /// Allocates a new heap with at most the given size in bytes.
+  pub fn new(bytes: Option<usize>) -> Option<Box<Self>> {
```

Comment:
This doc comment is incorrect. It allocates _exactly_ the given size in bytes. 

If we wanted it to allocate _at most_ the given size in bytes, then `new` should fall back to 1-TiB-and-go-down behavior when `new_exact` returns `None`

Reply:
Ah, yeah, originally I had it step down from `bytes.unwrap_or(1 << 40)`, but I thought it was undesirable if someone explicitly passed an amount of memory and it was silently downgraded.